managed-entitlements/src/managed_entitlements_checkout_borrow_sample.py

```diff
@@ -0,0 +1,147 @@
+    )
+
+def checkout_borrow_license(LicenseArn):
+    """Borrow a license for offline usage
+
+    The API returns an SignedToken which is a JWT token encoded using algorithm
+    PS384. The data is essentially the same returned as response of this object.
+    For more details: https://docs.aws.amazon.com/license-manager/latest/APIReference/API_CheckoutBorrowLicense.html
+    """
+
+    lm_client = boto3.client('license-manager', region)
+    return lm_client.checkout_borrow_license(
+        LicenseArn=LicenseArn,
+        Entitlements=[{
+            "Name":"Users",
+            "Value": "1",
+            "Unit":"Count"
+        }],
+        NodeId="MyNodeId",
+        DigitalSignatureMethod="JWT_PS384",
+        ClientToken=str(uuid.uuid4()).replace("-","")
+    )
+
+
+def create_cmk(desc='Customer Master Key'):
+    """Create a KMS Customer Master Key
+
+    The created CMK is a Customer-managed key stored in AWS KMS.
+    """
+
+    kms_client = boto3.client('kms', region)
+    response = kms_client.create_key(
+                                    Description=desc,
+                                    CustomerMasterKeySpec="RSA_4096",
+                                    KeyUsage="SIGN_VERIFY"
+                                )
+
+    return response['KeyMetadata']['Arn']
+
+def get_public_key(KeyId):
+    """Get CMK PublicKey
+    """
+
+    kms_client = boto3.client('kms', region)
+    response = kms_client.get_public_key(KeyId=KeyId)
+
+    return response['PublicKey']
+
+def main(command_line=None):
+    print("Start of the sample model for checkout borrow license")
+    # Initially, you should create an asymmetric key that License Manager uses to
+    # sign borrow license data. The algorithms are based on JWT standard and all
+    # major languages has public libraries to verify signatures. For this sample,
+    # we are using PyJwt
+    SignKey=create_cmk()
+    print(f"Key Created: {SignKey}")
+
+    # For offline verification public key should be stored safely in way that the Software
+    # can retrieve it. For more details how to manage your public key, please check
+    # KMS docs: https://docs.aws.amazon.com/kms/latest/APIReference/API_GetPublicKey.html
+    public_key=f"-----BEGIN PUBLIC KEY-----\n{base64.b64encode(get_public_key(SignKey)).decode()}\n-----END PUBLIC KEY-----\n"
+    print(f"Public Key: {public_key}")
+
+    # Creates a simple license for our example. We have an entitlement called Users
+    # and are indicating to License Manager that it should allow license borrowing
+    # by specifying BorrowConfiguration.
+    License=create_license(SignKey=SignKey)
+    LicenseArn=License["LicenseArn"]
+    print(f"License Created: {LicenseArn}")
+
+    # Borrowing a license. For real scenarios, it should be also accounted the
+    # distribution and potential leaks. Use borrowing specifying and validating
+    # NodeId if distribution is not safe.
+    signed_token=checkout_borrow_license(LicenseArn)["SignedToken"]
```

Comment:
Add comment here on the region of the key.

Reply:
Addressed in new revision